File: lib/core/rwanda_season.dart
/// Rwanda growing seasons from the current calendar date.
class RwandaSeason {
  static const seasonA = 'season_a';
  static const seasonB = 'season_b';
  static const seasonC = 'season_c';

  /// Season A: Sep–Jan, B: Feb–Jun, C: Jul–Aug
  static String forDate(DateTime date) {
    final m = date.month;
    if (m == 9 || m == 10 || m == 11 || m == 12 || m == 1) return seasonA;
    if (m >= 2 && m <= 6) return seasonB;
    return seasonC;
  }

  static String current() => forDate(DateTime.now());

  static String label(String seasonId) {
    switch (seasonId) {
      case seasonA:
        return 'Season A — Short rains (Sep–Jan)';
      case seasonB:
        return 'Season B — Long rains (Feb–Jun)';
      case seasonC:
        return 'Season C — Dry season (Jul–Aug)';
      default:
        return seasonId.replaceAll('_', ' ');
    }
  }

  static String shortAdvice(String seasonId) {
    switch (seasonId) {
      case seasonA:
        return 'Plant with the short rains; improve drainage on clay soils.';
      case seasonB:
        return 'Main season — apply basal fertilizer at planting and weed early.';
      case seasonC:
        return 'Dry period — use drought-tolerant crops or irrigation.';
      default:
        return '';
    }
  }
}

File: backend/app/rwanda_season.py
"""Rwanda agricultural seasons from calendar date (Kigali / national convention)."""
from __future__ import annotations

from datetime import date, datetime, timezone

SEASON_A = "season_a"  # Sep-Jan — short rains
SEASON_B = "season_b"  # Feb-Jun — long rains
SEASON_C = "season_c"  # Jul-Aug — dry season

SEASON_INFO: dict[str, dict[str, str]] = {
    SEASON_A: {
        "label": "Season A — Short rains (Sep–Jan)",
        "months": "September to January",
        "advice": "Plant at the start of the short rains; ensure drainage on heavy soils.",
    },
    SEASON_B: {
        "label": "Season B — Long rains (Feb–Jun)",
        "months": "February to June",
        "advice": "Main cropping window — complete basal fertilizer and timely weeding.",
    },
    SEASON_C: {
        "label": "Season C — Dry season (Jul–Aug)",
        "months": "July to August",
        "advice": "Favour drought-tolerant crops or irrigation; conserve soil moisture with mulch.",
    },
}


def rwanda_season_for_month(month: int) -> str:
    """Return season id for month 1-12."""
    if month in (9, 10, 11, 12, 1):
        return SEASON_A
    if month in (2, 3, 4, 5, 6):
        return SEASON_B
    return SEASON_C


def current_rwanda_season(when: date | datetime | None = None) -> str:
    if when is None:
        when = datetime.now(timezone.utc)
    if isinstance(when, datetime):
        month = when.month
    else:
        month = when.month
    return rwanda_season_for_month(month)


def season_label(season_id: str) -> str:
    return SEASON_INFO.get(season_id, {}).get("label", season_id.replace("_", " ").title())
